### PR TITLE
refactor: improve output structure for the parser

### PR DESCRIPTION
**Pull request by ghost**
BREAKING CHANGE:

* ` spectrum: Float32Array[] | Float32Array` is now `spectra: Float32Array[]`

### CODE REVIEW

File: .github/workflows/nodejs-ts.yml [hunk 1]
```diff
@@ -11,5 +11,5 @@ jobs:
     # Documentation: https://github.com/zakodium/workflows#nodejs-ci
     uses: zakodium/workflows/.github/workflows/nodejs.yml@nodejs-v1
     with:
-      node-version-matrix: '[14, 16]'
+      node-version-matrix: '[14, 16, 18]'
```
Comment (targos) [inline .github/workflows/nodejs-ts.yml]:
```suggestion
```

The default matrix contains all supported versions of Node.js. This also includes v19.x and can be updated in one place for all.

**Comment by targos** [comment]
On the commit message:

Please only use the exclamation mark if the message itself explains what the breaking changes are. Otherwise, like in this case, write the message like this:

```
refactor: summary of the changes

BREAKING CHANGE: explain what is broken for the changelog
```

File: src/maps.ts [hunk 2]
```diff
@@ -438,15 +438,12 @@ Spectrum flags are details to be aware of when reading the spectra.
...
-  lower: number,
-  higher: number,
-): WdfSpectrumFlags {
+export function getWdfSpectrumFlags(flag: bigint): WdfSpectrumFlags {
   return {
-    saturated: (lower & 1) !== 0,
-    error: (lower & 0b10) !== 0,
-    cosmicRay: (lower & 0b100) !== 0,
-    errorCode: higher >>> 31,
+    saturated: !!(flag & BigInt(1)),
```
Comment (targos) [inline src/maps.ts]:
```suggestion
    saturated: !!(flag & 1n),
```

why not use the bigint literal syntax?

File: tsconfig.json [hunk 3]
```diff
@@ -7,7 +7,8 @@
     "outDir": "lib",
     "sourceMap": true,
     "strict": true,
-    "target": "es2020"
+    "target": "es2020",
+    "noImplicitAny":true
```
Comment (targos) [inline tsconfig.json]:
```suggestion
    "noImplicitAny": true
```

**Comment by ghost** [comment]
Thanks for the help, I'll update with the improvements.

**Review by lpatiny — APPROVED** [review]
Just as a note we should take care returning BigInt to end users because they may be quite confused. For example it can not be stringified.

But if we have no choice it is like that.

```
JSON.stringify({ab: 1n})
VM187:1 Uncaught TypeError: Do not know how to serialize a BigInt
    at JSON.stringify (<anonymous>)
    at <anonymous>:1:6
```

**Comment by ghost** [comment]
Thanks! Yes the object returned has no bigint, it was only internally. I merged biologic because noticed you may be busy. 

It didn't make the npm publish though, both here and biologic. I think it was my mistake in both for using `refactor` in the title. 

Should I add a dummy commit with the breaking change @lpatiny  ?

**Comment by santimirandarp** [comment]
I tried but I think I did it wrong, I thought the PR message in top of the page was included and would bump the version, but it didn't and hence it didn't run npm-publish.

**Comment by targos** [comment]
@santimirandarp This mistake was not about using `refactor` in the title. It's that you committed with GitHub's default message, which didn't include the "BREAKING CHANGE:" text.
I see that you tried to fix it in https://github.com/cheminfo/wdf-parser/pull/14, but it still didn't include the breaking change text, so we ended up with this changelog: https://github.com/cheminfo/wdf-parser/releases/tag/v0.3.0
It's not a big deal, but please try to be careful about the commit message when you use the merge button. GitHub only does the right thing for us if there's a single commit in the pull request.

**Comment by ghost** [comment]
Yes, next time I will make sure the body and title of the merge commit make actual sense.

It seems that the `bigint` literal has trouble to be used in browsers required in analysis-ui. If I replace with BigInt(number) it passes the tests. 
I am trying to make sure this isn't just some testing restriction @targos, as the [support seems quite broad](https://caniuse.com/?search=bigint).

Maybe just going back to number is the best option.

**Pull request merged**